src/lib/DatePicker.svelte

```diff
@@ -127,95 +127,86 @@
     const maxDate = new Date(max.getFullYear(), max.getMonth(), max.getDate())
     return date >= minDate && date <= maxDate
   }
-  function shiftKeydown(e: KeyboardEvent) {
-    if (e.shiftKey && e.key === 'ArrowUp') {
-      setYear(year - 1)
-    } else if (e.shiftKey && e.key === 'ArrowDown') {
-      setYear(year + 1)
-    } else if (e.shiftKey && e.key === 'ArrowLeft') {
-      setMonth(month - 1)
-    } else if (e.shiftKey && e.key === 'ArrowRight') {
-      setMonth(month + 1)
-    } else {
-      return false
-    }
-    e.preventDefault()
-    return true
-  }
   function yearKeydown(e: KeyboardEvent) {
-    let shift = e.shiftKey || e.altKey
-    if (shift) {
-      shiftKeydown(e)
-      return
-    } else if (e.key === 'ArrowUp') {
+    if (e.key === 'ArrowUp') {
       setYear(year - 1)
+      e.preventDefault()
     } else if (e.key === 'ArrowDown') {
       setYear(year + 1)
+      e.preventDefault()
     } else if (e.key === 'ArrowLeft') {
       setMonth(month - 1)
+      e.preventDefault()
     } else if (e.key === 'ArrowRight') {
       setMonth(month + 1)
-    } else {
-      shiftKeydown(e)
-      return
+      e.preventDefault()
     }
-    e.preventDefault()
   }
   function monthKeydown(e: KeyboardEvent) {
-    let shift = e.shiftKey || e.altKey
-    if (shift) {
-      shiftKeydown(e)
-      return
-    } else if (e.key === 'ArrowUp') {
+    if (e.key === 'ArrowUp' || e.key === 'ArrowLeft') {
       setMonth(month - 1)
-    } else if (e.key === 'ArrowDown') {
+      e.preventDefault()
+    } else if (e.key === 'ArrowDown' || e.key === 'ArrowRight') {
       setMonth(month + 1)
-    } else if (e.key === 'ArrowLeft') {
+      e.preventDefault()
+    }
+  }
+
+  function shiftKeydown(e: KeyboardEvent) {
+    if (e.shiftKey && e.key === 'ArrowUp') {
+      setYear(year - 1)
+    } else if (e.shiftKey && e.key === 'ArrowDown') {
+      setYear(year + 1)
+    } else if (e.shiftKey && e.key === 'ArrowLeft') {
       setMonth(month - 1)
-    } else if (e.key === 'ArrowRight') {
+    } else if (e.shiftKey && e.key === 'ArrowRight') {
       setMonth(month + 1)
     } else {
-      shiftKeydown(e)
-      return
+      return false
     }
     e.preventDefault()
+    return true
   }
-  function keydown(e: KeyboardEvent) {
+
+  function calendarDayKeydown(e: KeyboardEvent) {
     let shift = e.shiftKey || e.altKey
-    if ((e.target as HTMLElement)?.tagName === 'SELECT') {
-      return
-    }
     if (shift) {
       shiftKeydown(e)
       return
+    } else if (e.key === ' ') {
+      dispatch('select')
+      e.preventDefault()
```

Comment:
^-- small change to enable the spacebar as a selection key (similar to how dropdowns/checkboxes/radio controls handle the spacebar)

Reply:
Good idea